### ISSUE TITLE
"Subscribe in Your Browser" uses a deprecated function not supported by Firefox

### ISSUE THREAD

**Issue by strugee** [gpodder/mygpo#229]
STR:

1. Sign in to GPodder in Firefox (I'm using Firefox Nightly, built July 6th, 2019, but AFAICT this should also be broken in Firefox stable)
2. Click the "Install" button in the "Subscribe in Your Browser" box on the lower-right of the page
3. Nothing appears to happen, and in the error console of dev tools I get this error: `TypeError: navigator.registerContentHandler is not a function`

According to MDN the function in question is [obsolete and hasn't existed in Firefox since Firefox 2](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerContentHandler). We want [`navigator.registerProtocolHandler()`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler) instead.

Thank you for maintaining GPodder <3

**Comment by SiqingYu** [comment]
I've tested on my platform, with Firefox 70.0b6 and local VM with TLS/HTTPS setup.

Firefox throws out an error: `SecurityError: Permission denied to add a protocol handler for application/vnd.mozilla.maybe.feed`

I guess Firefox has ditched support for adding a handler for RSS feed. There's no RSS scheme support in `navigator.registerProtocolHandler()`.

Reference:
* Navigator.registerProtocolHandler() - Web APIs | MDN
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler
* Adding feed readers to Firefox - Mozilla | MDN
https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/2/Adding_feed_readers_to_Firefox

**Comment by stefankoegl** [comment]
Thanks for the report! I've merged and deployed #230.

**Comment by strugee** [comment]
Thank you for your hard work on GPodder/this issue @stefankoegl and @KeironO! I appreciate this service very much :-)

**Comment by strugee** [comment]
Actually, turns out I get the same SecurityError reported in https://github.com/gpodder/mygpo/pull/230#issuecomment-531812215...

**Comment by SiqingYu** [comment]
As an RSS user, I don't bind rss feeds to specific applications like Inoreader or Pocket Casts. The browser can't distinguish podcasts from usual blog feeds by plain XML link.

**Comment by strugee** [comment]
@SiqingYu I'm not sure I understand what you're trying to say. Can you rephrase?

Also looking at the MDN documentation, looks like the best we can do is register a protocol handler for `news:`. I don't know how much that's used in the wild though; anecdotally I've usually just seen links to RSS feeds which that won't work for...

**Comment by SiqingYu** [comment]
@strugee Actually, the `news` URL scheme is not used for RSS/podcasts, but USENET news instead.
RSS/podcasts links are HTTP/HTTPS URL schemes, identified by files ended with XML/OPML extension.

References:

- RFC 5538 - The 'news' and 'nntp' URI Schemes
https://tools.ietf.org/html/rfc5538

**Comment by SiqingYu** [comment]
https://github.com/gpodder/mygpo/pull/230#issuecomment-532975068

**Comment by strugee** [comment]
Nice catch. I _knew_ there was something funky about that... guess this is what I get for commenting on issues after midnight :P
